src/app/components/bookmark/bookmark.component.ts

```diff
@@ -43,27 +44,38 @@ export class BookmarkComponent {
   public pageNumber = 0;
 
   public isPersistenceActive = false;
+  public currentCollections = '';
 
   @Output() public actions: Subject<{ action: string, id: string, geometry?: any }> = new Subject<any>();
 
   @ViewChild(MatPaginator, { static: false }) public paginator: MatPaginator;
 
   constructor(
     private bookmarkService: ArlasBookmarkService,
-    private dialog: MatDialog
+    private dialog: MatDialog,
+    private startupService: ArlasStartupService
   ) {
+    this.currentCollections = Array.from(this.startupService.collectionsMap.keys()).join(',');
     // Init component with data from persistence server, if defined and server is reachable
     if (this.bookmarkService.dataBase instanceof BookmarkPersistenceDatabase) {
-          this.isPersistenceActive = true;
-          this.bookmarkService.setPage(this.pageSize, this.pageNumber);
-          this.getBookmarksList();
-          (this.bookmarkService.dataBase as BookmarkPersistenceDatabase).dataChange
-            .subscribe((data: { total: number, items: BookMark[] }) => {
-              this.resultsLength = data.total;
-              this.bookmarks = data.items;
-            });
+      this.isPersistenceActive = true;
+      this.bookmarkService.setPage(this.pageSize, this.pageNumber);
+      this.getBookmarksList();
+      (this.bookmarkService.dataBase as BookmarkPersistenceDatabase).dataChange
+        .subscribe((data: { total: number, items: BookMark[] }) => {
+          this.resultsLength = data.total;
+          this.bookmarks = data.items.filter(bk => {
```

Comment:
if there is a repeated filter, it could also be factorized in a dedicated function?